java/statestore/src/main/java/sleeper/statestore/transactionlog/DynamoDBTransactionLogStore.java

```diff
@@ -119,6 +120,29 @@ public Stream<TransactionLogEntry> readTransactionsAfter(long lastTransactionNum
                 .map(this::readTransaction);
     }
 
+    @Override
+    public void deleteTransactionsAtOrBefore(long transactionNumber) {
+        streamPagedItems(dynamo, new QueryRequest()
+                .withTableName(logTableName)
+                .withConsistentRead(true)
+                .withKeyConditionExpression("#TableId = :table_id AND #Number <= :number")
+                .withExpressionAttributeNames(Map.of("#TableId", TABLE_ID, "#Number", TRANSACTION_NUMBER))
+                .withExpressionAttributeValues(new DynamoDBRecordBuilder()
+                        .string(":table_id", sleeperTableId)
+                        .number(":number", transactionNumber)
+                        .build())
+                .withReturnConsumedCapacity(ReturnConsumedCapacity.TOTAL))
+                .forEach(item -> {
+                    long itemTransactionNumber = getLongAttribute(item, TRANSACTION_NUMBER, 0L);
+                    if (item.get(BODY_S3_KEY) != null) {
+                        LOGGER.info("Deleting transaction body in S3 for transaction number {} and table {}", itemTransactionNumber, sleeperTableId);
+                        s3.deleteObject(new DeleteObjectRequest(dataBucket, item.get(BODY_S3_KEY).getS()));
+                    }
+                    LOGGER.info("Deleting transaction from DynamoDB for transaction number {} and table {}", itemTransactionNumber, sleeperTableId);
```

Comment:
These logs seem very noisy, there could be thousands of transactions to be deleted. Maybe we should just summarise at the end?